packages/xdl/src/Webpack.ts

```diff
@@ -363,3 +415,133 @@ async function getWebpackConfigEnvFromBundlingOptionsAsync(
     ...(options.webpackEnv || {}),
   };
 }
+
+async function startNextJsAsync({ projectRoot, port, dev, expoConfig, onFinished }) {
+  let next;
+  try {
+    const { exp } = await ProjectUtils.readConfigJsonAsync(projectRoot);
+    next = require(ConfigUtils.resolveModule('next', projectRoot, exp));
+  } catch {
+    throw new XDLError('NEXTJS_NOT_INSTALLED', 'Next.js is not installed in your app.');
+  }
+
+  // Build first if in production mode.
+  // https://nextjs.org/docs#custom-server-and-routing
+  if (!dev) {
+    await bundleNextJsAsync({ projectRoot, expoConfig });
+  }
+
+  const app = next({
+    dev,
+    dir: projectRoot,
+    conf: _createNextJsConf({
+      projectRoot,
+      expoConfig,
+    }),
+  });
+  const handle = app.getRequestHandler();
+
+  app.prepare().then(() => {
+    const server = express();
+    server.get('*', handle);
+
+    webpackDevServerInstance = server.listen(port, err => {
+      if (err) {
+        throw new XDLError(
+          'EXPRESS_FAIL_TO_START',
+          `Express server failed to start: ${err.toString()}`
+        );
+      }
+      onFinished(webpackDevServerInstance);
+    });
+  });
+}
+
+async function bundleNextJsAsync({ projectRoot, expoConfig }) {
+  let nextBuild;
+  try {
+    const { exp } = await ProjectUtils.readConfigJsonAsync(projectRoot);
+    nextBuild = require(ConfigUtils.resolveModule('next/dist/build', projectRoot, exp)).default;
+  } catch {
+    throw new XDLError(
+      'NEXTJS_NOT_INSTALLED',
+      'Next.js (or its build component) is not installed in your app.'
+    );
+  }
+
+  await nextBuild(projectRoot, _createNextJsConf({ projectRoot, expoConfig }));
+}
+
+function _createNextJsConf({ projectRoot, expoConfig }) {
```

Comment:
```suggestion
function _createNextJsConfig({ projectRoot, expoConfig }) {
```
(In general, we avoid abbreviations in code, but `config` is so well established as an abbreviation of "configuration" that it's OK. `conf` is not. 😄)